rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -996,6 +996,54 @@ private void logUpdateCollections(Transaction rskTx) {
         eventLogger.logUpdateCollections(sender);
     }
 
+    protected void processValidationFailure(Federation proposedFederation) {
+        eventLogger.logCommitFederationFailure(rskExecutionBlock, proposedFederation);
+        logger.warn("[processValidationFailure] Proposed federation validation failed so svp values will be cleared.");
+        clearSvpValues();
+    }
+
+    private void clearSvpValues() {
+        federationSupport.clearProposedFederation();
+
+        String methodName = "[clearSvpValues]";
+
+        Optional<Sha256Hash> svpFundTxHashUnsigned = provider.getSvpFundTxHashUnsigned();
+        if (svpFundTxHashUnsigned.isPresent()) {
+            logger.warn("{} Fund tx {} was never registered.", methodName, svpFundTxHashUnsigned.get());
+            provider.setSvpFundTxHashUnsigned(null);
+            return;
+        }
+
+        Optional<BtcTransaction> svpFundTxSigned = provider.getSvpFundTxSigned();
+        if (svpFundTxSigned.isPresent()) {
+            logger.warn("{} Spend tx was never created. Fund tx hash: {}", methodName, svpFundTxSigned.get().getHash());
+            provider.setSvpFundTxSigned(null);
+            return;
```

Comment:
I noticed we return early since I am guessing it does not make sense to check one value given the SVP timeline. But should we not try to clear all of them just in case? Just to be in the safe side

Reply:
one of the svp flow focus is not to have more than one of these values simultaneously. Each value represents one different stage of the process.

Reply:
Just to clarify, with the new changes present now we are actually implementing what I was mentioning above. I just want to make sure you are aware of this. Otherwise, LGTM